src/mscorlib/shared/System/IO/Path.Unix.cs

```diff
@@ -189,6 +189,14 @@ public static string GetPathRoot(string path)
             return IsPathRooted(path) ? PathInternal.DirectorySeparatorCharAsString : String.Empty;
         }
 
+        public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
+        {
+            if (PathInternal.IsEffectivelyEmpty(new string(path.ToArray())))
```

Comment:
You should just check `path.IsEmpty` instead of allocating a new string just for the check.

Reply:
Also, I am not sure whether it is desirable to throw here at all for empty span. It may be more appropriate to just return empty span instead of throwing.

Reply:
Or maybe add an override of PathIsEffectivelyEmpty that takes a Span in the Unix implementation as well. Cause I see you're adding one for Windows.